keps/sig-apps/19-Graduate-CronJob-to-Stable/README.md

```diff
@@ -0,0 +1,605 @@
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: `CronJobControllerV2`
+    - Components depending on the feature gate: `kube-controller-manager`
+
+* **Does enabling the feature change any default behavior?**
+  No.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  No changes are expected.
+
+* **Are there any tests for feature enablement/disablement?**
+  There are unit tests at minimum that verify full correctness of the new controller. Additionally, we want to ensure the new controller passes all the existing e2e tests for the current controller.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  The only time when this might fail is when leader election for kube-controller-manager ends up assigned to two separate instances. In this case the two controller might end up fighting over the cronjob resource.
+
+* **What specific metrics should inform a rollback?**
+  None.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  I do not believe this is applicable.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs,
+fields of API types, flags, etc.?**
+  batch/v1beta1.CronJob is already deprecated.
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  `cronjob_job_creation_skew` and `cronjob_controller_rate_limiter_use` accompanied with a set of queue-related metrics the old controller is not using.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine
+the health of the service?**
+  - [x] Metrics
+    - Metric name: `cronjob_controller_rate_limiter_use`
+  	- Metric name: `cronjob_job_creation_skew`
+  	- Metric name: `workqueue_depth`
+  	- Metric name: `workqueue_retries`
+  	- Metric name: `workqueue_adds_total`
+    - Components exposing the metric: `kube-controller-manager`
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  TBD
+
+* **Are there any missing metrics that would be useful to have to improve observability
+of this feature?**
+  No.
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  No.
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+  - WATCH cronjobs
+  - estimated throughput TBD
+  - originating component(s) kube-controller-manager
+
+* **Will enabling / using this feature result in introducing new API types?**
+  No.
+
+* **Will enabling / using this feature result in any new calls to the cloud
+provider?**
+  Yes.
```

Comment:
No (you're not talking to GCE/AWS/.... etc)

Reply:
Sorry, my bad. Fixed.